docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/spryker-elastic-computing.md

```diff
@@ -0,0 +1,316 @@
+
+
+
+
+This can be achieved with elastic batch concept introduced in `spryker/data-import` module. The following instruction will explain the integration details of elastic batch using the glossary data import as an example.
+
+> :info: Similar integration must be applied for all project data importers.
+
+1. In `Pyz\Zed\DataImport\DataImportConfig`, add the following configuration:
+
+* `BulkWriteGradualityFactor`: estimates an upper limit of memory that can be safely utilized by operations. A bigger value enables a more precise approximation but requires more iterations. A smaller value gives a less precise approximation but requires less iterations.
+
+* `BulkWriteMemoryThresholdPercent`: defines a margin of PHP memory limit configured in PHP.
+
+
+```php
+/**
+ * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+ */
+class DataImportConfig extends SprykerDataImportConfig
+{
+    ...
+        
+    /**
+     * @return int
+     */
+    public function getBulkWriteGradualityFactor(): int
+    {
+        return 5;
+    }
+
+    /**
+     * @return int
+     */
+    public function getBulkWriteMemoryThesoldPercent(): int
+    {
+        return 95;
+    }
+}
+```
+
+2. In `src/Pyz/Zed/DataImport/Business/DataImportBusinessFactory.php`, adjust the creation of data importer so it’s new version will
+- use `DataSetStepBrokerElasticBatchTransactionAware` data set step broker and
+- the writer steps receive the `MemoryAllocatedElasticBatch`
+```php
+/**
+ * @method \Pyz\Zed\DataImport\DataImportConfig getConfig()
+ * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+ * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
+ * @SuppressWarnings(PHPMD.ExcessiveClassLength)
+ */
+class DataImportBusinessFactory extends SprykerDataImportBusinessFactory
+{
+  ...
+  protected function createGlossaryImporter(
+        DataImportConfigurationActionTransfer $dataImportConfigurationActionTransfer 
+        ): DataImporterInterface
+    {
+      $dataImporter = $this->getCsvDataImporterFromConfig(
+            $this->getConfig()->buildImporterConfigurationByDataImportConfigAction($dataImportConfigurationActionTransfer),
+        );
+      $dataSetStepBroker = $this->createElasticBatchTransactionAwareDataSetStepBroker(GlossaryWriterStep::BULK_SIZE);
+      $dataSetStepBroker
+        ->addStep($this->createLocaleNameToIdStep(GlossaryWriterStep::KEY_LOCALE))
+        ->addStep(new GlossaryWriterStep($this->createMemoryAllocatedElasticBatch()));
+
+        $dataImporter->addDataSetStepBroker($dataSetStepBroker);
+
+        return $dataImporter;
+    }
+  ...
+}
+```
+
+3. Update the writer steps (`Pyz\Zed\DataImport\Business\Model\Glossary\GlossaryWriterStep`) as it is show in the code below by adjusting the execute method to flush `MemoryAllocatedElasticBatch` when it is full.
```

Comment:
Why is `Pyz\Zed\DataImport\Business\Model\Glossary\GlossaryWriterStep` in paratheses? Is it an example?

Reply:
Yes, this is an example. We are specifying the actual file to be changed with the following block of code.